src/backend/tty.rs

```diff
@@ -1102,13 +1104,114 @@ impl Tty {
             }
         };
 
+        let overlay_cursor = niri
+            .output_state
+            .get(output)
+            .and_then(|state| state.pending_screencopy.as_ref())
+            .map_or(true, |s| s.overlay_cursor());
+
         // Render the elements.
-        let elements = niri.render::<TtyRenderer>(&mut renderer, output, true);
+        let elements = niri.render::<TtyRenderer>(&mut renderer, output, overlay_cursor);
 
         // Hand them over to the DRM.
         let drm_compositor = &mut surface.compositor;
         match drm_compositor.render_frame::<_, _>(&mut renderer, &elements, [0.; 4]) {
             Ok(res) => {
+                if let Some(screencopy) = niri
```

Comment:
Is there any particular reason why you put this code in both tty.rs and winit.rs? I'd put it in niri.rs in the end of `redraw()`, together with the call to `self.render_for_screen_cast()` which does a conceptually similar thing. It always renders separately, too, so you can always use the correct `overlay_cursor` setting without stepping on the real render's toes. And there's no code duplication between winit and tty.

Yes that means it won't optimize by blitting when appropriate, but I think for the first iteration that's fine (especially for taking 1 screenshot use case), and later I'll try to solve it for both PipeWire and screencopy together.

Reply:
The main reason i put it in both is because the bodies are not identical (i.e. `res` is of a different type in each, the `SyncPoint` is gotten from a different place, and of course on TTY there is the dmabuf implementation). I didn't realize there was an appropriate unified place to put this, though, and i'll see what i can do on this regard. Having a separate rendering function to hook into is most definitely what i want in reality, though, so i'll try to implement it there.

Reply:
I've implemented it in a similar manner to `render_for_screen_cast`. Now, screencopy is handled at the end of the main render function.